src/main/scala/zio/package.scala

```diff
@@ -317,11 +321,22 @@ package object zmx extends MetricsDataModel with MetricsConfigDataModel {
           }
         )
 
-      private val untilNCollected                                                         = Schedule.recurUntil[List[Metric[_]]](_.size == config.bufferSize)
+      // TODO:
+      // the untilNCollected schedule is combined with a very short timed schedule
+      // -> otherwise `poll.repeat(untilCollected)` would never block
+      // -> it seems that even TestClock.adjust does block then
+      // -> maybe a different mechanism should be used to couple the ring buffer and aggregator (STM FTW?)
+      private val untilNCollected =
+        Schedule.recurUntil[List[Metric[_]]](_.size == config.bufferSize) <* Schedule.spaced(10.milliseconds)
+
+      // given a function List[Metric[_]] => Task[List[Long]]:
+      // -> returns a Task[List[Long]]
+      // -> this task polls until the aggregator at least contains the configured buffer size of metrics
+      // -> then it applies the polled list to the given function
       private[zio] val collect: (List[Metric[_]] => Task[List[Long]]) => Task[List[Long]] =
         f => {
           for {
-            _ <- poll.repeat(untilNCollected).provideLayer(Clock.live)
+            _ <- poll.repeat(untilNCollected).provideSome((_: Any) => Has(clock))
```

Comment:
This can be `provide(Has(clock))`. If the environment type you are providing is `Any` you aren't providing some of the environment, you are providing all of it. What you have here works but using `provide` is cleaner and clarifies intent.